samza-core/src/main/java/org/apache/samza/operators/impl/OutputOperatorImpl.java

```diff
@@ -69,4 +69,14 @@ protected void handleClose() {
   protected OperatorSpec<M, Void> getOperatorSpec() {
     return outputOpSpec;
   }
+
+  @Override
+  protected long handleWatermark(Watermark inputWatermark,
+      MessageCollector collector,
+      TaskCoordinator coordinator) {
+    if (outputOpSpec.getOpCode() == OperatorSpec.OpCode.PARTITION_BY) {
+      inputWatermark.propagate(outputStream.getStreamSpec().toSystemStream());
```

Comment:
Yup, completely agree. https://issues.apache.org/jira/browse/SAMZA-1351 tracks this.